silverCrawler/silverCrawler-war/src/main/java/org/silverpeas/components/silvercrawler/control/SilverCrawlerSessionController.java

```diff
@@ -523,6 +503,27 @@ public boolean checkRWSettingsAccess(boolean throwException)
     return true;
   }
 
+  private boolean checkIsAdmin(final boolean throwException, final String[] userRoles)
+      throws SilverCrawlerForbiddenActionException {
+    boolean isAdmin = false;
+    for (String userRole : userRoles) {
+      if ("admin".equals(userRole)) {
+        isAdmin = true;
+        break;
+      }
+    }
+    if (!isAdmin) {
+      if (throwException) {
+        throw new SilverCrawlerForbiddenActionException(CHECK_RWSETTINGS_ACCESS,
+            SilverpeasException.ERROR,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "SilverpeasException"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)